src/conn.ts

```diff
@@ -225,8 +223,14 @@ export class Conn {
         } else {
           data = funcReturn;
         }
-        isCanceled = data === false;
-        if (data !== undefined && data !== false) {
+        if (!isCanceled) { // Wait for the first occurrence 
+          isCanceled = data === false;
+        } else {
+          if (data === true) {
+            isCanceled = false; // Allow following middlewares to un cancel packet that have been canceled already
+          }
+        }
```

Comment:
actually, why is this code here twice? maybe the middleware application could be made into a utility function to avoid code duplication?
```suggestion
        isCanceled = isCanceled ? data !== true : data === false;
```